src/Views/Games/Sudoku/Board.js

```diff
@@ -44,10 +41,29 @@ function Board({ boardState }) {
     const [validChoices, setValidChoices] = useState([]);
 
     const [selectedTile, setSelectedTile] = useState(null);
-    // To check the board we need to check rows, columns, and squares. Returns true if the board is complete, false otherwise
-    function checkBoard() {}
+
+    function checkBoard() {
+        for (let i = 0; i < 9; i++) {
+            let row = getRow(i);
+            let column = getColumn(i);
+            let square = getSquare(i);
+            for (let j = 0; j < 9; j++) {
```

Comment:
This inner loop is unnecessary. Use sets and set operations instead. What would the union of all the rows, columns, and squares look like if it's complete vs incomplete